codex/codex.nim

```diff
@@ -194,7 +196,23 @@ proc new*(
     .withTcpTransport({ServerFlags.ReuseAddr})
     .build()
 
-  var cache: CacheStore = nil
+  var
+    cache: CacheStore = nil
+    taskpool: Taskpool
+
+  try:
+    if config.numThreads == 0:
+      taskpool = Taskpool.new(numThreads = min(countProcessors(), 16))
+    elif config.numThreads < 2:
+      raise newException(
```

Comment:
I would use a `raiseAssert` here instead.

Reply:
Actually, I think this is better checked in the `conf.nim` so that the invalid value doesn't get passed the cmd validation.